com.unity.render-pipelines.core/Editor/Lighting/SRPLensFlareElementEditor.cs

```diff
@@ -0,0 +1,490 @@
+                    Rect imgRect = new Rect(position.x - 35.0f + 15.0f, position.y + imgOffY + GUIStyle.none.lineHeight, imgWidth, imgWidth);
+                    EditorGUI.DrawTextureTransparent(imgRect, lensFlareProp.objectReferenceValue as Texture, ScaleMode.ScaleToFit, usedAspectRatio);
+                }
+            }
+            Rect rect = m_CurrentRect;
+            if (isFoldOpenedProp.boolValue && flareType != SRPLensFlareType.Circle && flareType != SRPLensFlareType.Polygon)
+            {
+                m_CurrentRect.y += 1.5f * 35.0f;
+            }
+            EditorGUI.BeginProperty(new Rect(rect.x, rect.y, rect.width, 2.0f * rect.height), label, property);
+
+            float lineHeight = EditorGUIUtility.singleLineHeight + EditorGUIUtility.standardVerticalSpacing;
+
+            Color tmpCol;
+            bool tmpBool;
+            float tmp;
+            int iTmp;
+            Vector2 tmpVec2;
+            if (EditorGUI.BeginFoldoutHeaderGroup(new Rect(position.x, position.y, position.width, GUIStyle.none.lineHeight), isFoldOpenedProp.boolValue, EditorGUIUtility.TrTextContent("Lens Flare Element")))
+            {
+                rect = GetNextRect();
+                EditorGUI.TextArea(rect, "Type", EditorStyles.boldLabel);
+                {
+                    rect = GetNextRect();
+                    SRPLensFlareType newType;
+                    SRPLensFlareType typeValue = (UnityEngine.SRPLensFlareType)flareTypeProp.enumValueIndex;
+                    if ((newType = ((SRPLensFlareType)(EditorGUI.EnumPopup(rect, Styles.flareType, typeValue)))) != typeValue)
+                        flareTypeProp.enumValueIndex = (int)newType;
+
+                    if (newType == SRPLensFlareType.Image)
+                    {
+                        Texture tmpTex;
+                        rect = GetNextRect();
+                        if ((tmpTex = (EditorGUI.ObjectField(rect, Styles.flareTexture, lensFlareProp.objectReferenceValue, typeof(Texture), false) as Texture)) != (lensFlareProp.objectReferenceValue as Texture))
+                        {
+                            lensFlareProp.objectReferenceValue = tmpTex;
+                            lensFlareProp.serializedObject.ApplyModifiedProperties();
+                        }
+                    }
+                    else if (newType == SRPLensFlareType.Circle || newType == SRPLensFlareType.Polygon)
+                    {
+                        SerializedProperty fallOffProp = property.FindPropertyRelative("fallOff");
+                        SerializedProperty edgeOffsetProp = property.FindPropertyRelative("edgeOffset");
+                        SerializedProperty sdfRoundnessProp = property.FindPropertyRelative("sdfRoundness");
+                        SerializedProperty sideCountProp = property.FindPropertyRelative("sideCount");
+                        SerializedProperty inverseSDFProp = property.FindPropertyRelative("inverseSDF");
```

Comment:
same here: this should be cached, not done at each OnGUI()

Reply:
Currently they are relative property, relative to property. Property is only accessible from OnGUI();